pkg/consensus/consensus.go

```diff
@@ -274,7 +274,7 @@ func (s *service) Start() {
 
 // Shutdown implements the Service interface.
 func (s *service) Shutdown() {
-	if s.started.Load() {
+	if s.started.CAS(true, false) {
```

Comment:
Is it enough? It depends on whether we allow `Start()` recall for the same consensus service instance. Currently we don't have such code, but the second call to `Start()` after `Shutdown()` will fail due to the fact that `s.quit` is closed. I'd prefer to use another atomic bool variable (something like `s.finished`) and shutdown service only if `s.finished` is false.

Reply:
> It depends on whether we allow Start() recall for the same consensus service instance

Yes, that's true. I thought about adding an explicit note about that in comments and maybe now it's time to do that.